### DISCUSSION TITLE
How to make a node terminate or become complete after the first parallel state becomes done?

### DISCUSSION THREAD

**Discussion by mattmazzola** [statelyai/xstate#4136] [General] [answered]
I have a process to convert uploaded data as follows:

1: The user submits a file to form
2: File is uploaded to storage in container A
3: Background process watching storage, is triggered, performs transformation on uploaded File A and saves a new File B to container B

The application only gets upload response from request to container A. It does not know about the background process and doesn't have a way to know when the transformation is completed and new Blob is stored to container B
I don't expect high frequency usage so cheap solution was polling.
After the response from upload to A, I am polling container B and assume any new blobs are the transformed blob.

Because of the polling I also need a timer to expire after a certain time and prevent waiting forever.

I have tried to model this as a State machine using Parallel states as shown here:

![blob_upload](https://github.com/statelyai/xstate/assets/2856501/41c644a9-5337-43c1-baa0-ab7feea062f2)

## Issue:
Even after I click "after 5000ms" to force the timer to expire, the machine does not go to the Final state.
It waits until I also click "blobFound" so that BOTH of the parallel states are in final nodes before consider the parent node done.

https://stately.ai/registry/editor/06e50f12-42c1-4de6-976a-e3c189930127?machineId=ca7df814-7f0e-4655-b216-1e6152c22223&mode=Design

## Expectation:
What I want is for the machine to go to the bottom Final state after the First parallel state reaches a termina state.
This would be like [Promise.race()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race)

Is this possible in XState/Stately.AI?

**Comment by ChrisShank** [comment, marked as answer]
I probably wouldn't use parallel states here if you dont need to https://stately.ai/viz/b00c4d23-fddd-40c9-a387-5fba43f31ee1

**Reply by mattmazzola** [reply]
Oh, I didn't know it was possible to transition from a state inside the Processing child machine to a state at the root machine (Process Complete)
I think this is simpler and will work for that I need. Thanks!
